torch/nn/quantized/modules/functional_modules.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from torch._ops import ops
+
+
+_VALID_OP_NAMES = {
+    'add',
+    'cat'
+}
+
+class FloatFunctional(torch.nn.Module):
+    def __init__(self, op_name):
+        super(FloatFunctional, self).__init__()
+        if op_name not in _VALID_OP_NAMES:
+            raise NotImplementedError("{} not supported.".format(op_name))
+        self.observer = torch.nn.Identity()
+        self.op_name = op_name
+
+    def forward(self, x):
+        raise RuntimeError("FloatFunctional is not intended to use the " +
+                           "'forward'. Please use the underlying operation")
+
+    def add(self, x, y):
+        # type: (Tensor, Tensor) -> Tensor
+        assert self.op_name == 'add', \
+            "Running add while initialized as {}".format(self.op_name)
+        r = torch.add(x, y)
+        self.observer(r)
```

Comment:
this might not work with QAT, because I think observer returns a value override in that case - cc @jerryzh168 . Maybe put a TODO now and address it in a separate diff?

Reply:
Good catch!, we will need to return a value for QAT.  @jerryzh168 : Even the existing code has a problem: We are calling fake quant in the forward hook and this needs to change numerics, which is not allowed.

Reply:
So should that be `r = self.observer(r)`?

Reply:
QAT observers return overrides

Reply:
Does that mean that QAT and PTQ observers have different semantics?

Reply:
That's the current observer semantics, if non-None is returned, it overrides the return: https://github.com/pytorch/pytorch/blob/77c08aa46c3f3460b95b89cbe357b99180bc824d/torch/nn/modules/module.py#L549-L551

Reply:
Got it! I will add a TODO, and can send a diff later on, while bringing @jerry39213gh into the loop

Reply:
This should be a very simple diff: https://github.com/pytorch/pytorch/pull/24194. I just sent it out. Also, it should be @jerryzh168